File: src/demos/ps/dungeons/Dezo_cave3.java
package demos.ps.dungeons;

import demos.ps.PSDungeon;
import demos.ps.oo.PSGame;
import demos.ps.oo.PSGame.Chest;
import demos.ps.oo.PSGame.Planet;
import demos.ps.oo.PSGame.Trapped;
import demos.ps.oo.PSLibEnemy.PS1Enemy;

public class Dezo_cave3 {

	public static void startmap() {
		PSDungeon dungeon = PSGame.currentDungeon;
		dungeon.setRandomEnemies(0, new PS1Enemy[]{PS1Enemy.WT_DRAGN});
		dungeon.startDungeon();
	}
	
	public static void chest() {
		PSGame.chestFlag(Chest.DEZO_CAVE3_CHEST, 0, Trapped.EXPLOSION, null);
	}
	
	public static void exit() {
		PSGame.mapswitch(Planet.DEZORIS,145,70);
	}
	public static void nextArea() {
		PSGame.mapswitch(Planet.DEZORIS,143,60);
	}

}

File: src/demos/ps/dungeons/Dezo_cave3.ts
/**
 * Dezo_cave3 - Dungeon Script
 * TypeScript port of Dezo_cave3.java
 */

import { PSGame } from '../PSGame';
import { Chest, Trapped } from '../game/GameData';
import { Planet } from '../game/City';
import { PS1Enemy } from '../game/PSLibEnemy';
import { MainEngine } from '../../../core/MainEngine';

export class Dezo_cave3 {

  public static async startmap(): Promise<void> {
    MainEngine.setScriptContext(Dezo_cave3);
    const dungeon = PSGame.getCurrentDungeonInstance();
    if (dungeon) {
      dungeon.setRandomEnemies(0, [PS1Enemy.WT_DRAGN]);
      await dungeon.startDungeon();
    }
  }

  public static async chest(): Promise<void> {
    await PSGame.chestFlag(Chest.DEZO_CAVE3_CHEST, 0, Trapped.EXPLOSION, null);
  }

  public static async exit(): Promise<void> {
    await PSGame.mapswitchToPlanet(Planet.DEZORIS, 145, 70);
  }

  public static async nextArea(): Promise<void> {
    await PSGame.mapswitchToPlanet(Planet.DEZORIS, 143, 60);
  }
}
